lib/Serverless.js

```diff
@@ -100,21 +99,29 @@ class Serverless {
 
     // populate variables after --help, otherwise help may fail to print
     // (https://github.com/serverless/serverless/issues/2041)
-    return this.variables.populateService(this.pluginManager.cliOptions).then(() => {
-      // merge arrays after variables have been populated
-      // (https://github.com/serverless/serverless/issues/3511)
-      this.service.mergeArrays();
-
-      // populate function names after variables are loaded in case functions were externalized
-      // (https://github.com/serverless/serverless/issues/2997)
-      this.service.setFunctionNames(this.processedInput.options);
-
-      // validate the service configuration, now that variables are loaded
-      this.service.validate();
+    return this.variables
+      .populateService(this.pluginManager.cliOptions)
+      .then(() => {
+        // merge arrays after variables have been populated
+        // (https://github.com/serverless/serverless/issues/3511)
+        this.service.mergeArrays();
+
+        // populate function names after variables are loaded in case functions were externalized
+        // (https://github.com/serverless/serverless/issues/2997)
+        this.service.setFunctionNames(this.processedInput.options);
+
+        // initialize hooks
+        return BbPromise.mapSeries(this.pluginManager.getHooks(['initialize']), ({ hook }) =>
+          hook()
```

Comment:
I would rather enclose it within `pluginManger.run` (as lifecycle events are processed purely in its context). We should also avoid non standard Bluebird functions, to make future migration to native promises (and async/await) easier.

Reply:
There's no `Promise.reduce`, but same can be achieved neatly via `[].reduce` I think